PowerPointLabs/PowerPointLabs/SyncLab/ObjectFormats/ThreeDRotationEffectFormat.cs

```diff
@@ -0,0 +1,99 @@
+        {
+            ThreeDFormat threeD = formatShape.ThreeD;
+            
+            // equality check for floating point numbers
+            return Math.Abs(threeD.RotationX) > TOLERANCE
+                   || Math.Abs(threeD.RotationY) > TOLERANCE
+                   || Math.Abs(threeD.RotationZ) > TOLERANCE
+                   || Math.Abs(threeD.FieldOfView) > TOLERANCE
+                   || threeD.Perspective == MsoTriState.msoTrue;
+        }
+
+        public override void SyncFormat(Shape formatShape, Shape newShape)
+        {
+            if (!Sync(formatShape, newShape))
+            {
+                Logger.Log(newShape.Type + " unable to sync 3DRotation Format");
+            }
+        }
+
+        public override Bitmap DisplayImage(Shape formatShape)
+        {
+            Bitmap image = GraphicsUtil.ShapeToBitmap(formatShape);
+            return image;
+        }
+
+        private static bool Sync(Shape formatShape, Shape newShape)
+        {
+            ThreeDFormat source = formatShape.ThreeD;
+            ThreeDFormat dest = newShape.ThreeD;
+            
+            try
+            {
+                if (source.PresetThreeDFormat == MsoPresetThreeDFormat.msoPresetThreeDFormatMixed)
+                {
+                    // skip setting PresetCamera if Mixed. setting it throws an error
+                    if (source.PresetCamera != MsoPresetCamera.msoPresetCameraMixed)
+                    {
+                        dest.SetPresetCamera(source.PresetCamera);
+                    }
+                    // set FieldOfView anyway, PresetCamera doesn't seem to set it
+                    dest.FieldOfView = source.FieldOfView;
+
+                    dest.RotationX = source.RotationX;
+                    dest.RotationZ = source.RotationZ;
+                    dest.RotationY = source.RotationY;
+
+                    // set perspective only if it is different,
+                    // setting the same perspective applies an unknown change to the lighting of the shape
+                    // this change visible to the eye, but we cannot undo it
+                    if (dest.Perspective != source.Perspective)
+                    {
+                        dest.Perspective = source.Perspective;
+                    }
+                }
+                else
+                {
+                    dest.SetThreeDFormat(source.PresetThreeDFormat);
+                }
+
+
+                // ThreeDFormat.Z must be between -4000 & 4000 exclusive.
+                // when source.Z > 4000 or source.Z < - 4000, it actually means 0
+                float nearestZ = source.Z;
+                nearestZ = nearestZ > 4000 ? 0f : nearestZ;
+                nearestZ = nearestZ < -4000 ? 0f : nearestZ;
+                dest.Z = nearestZ;
+                
+                dest.ProjectText = source.ProjectText;
+
+            }
+            catch (Exception)
+            {
+                return false;
```

Comment:
Log this exception?

Reply:
Hmm.. 
We actually skip logging in the static sync method for all Format classes thus far & only log in the public instance method. Perhaps I should have gotten rid of the static sync method back when I did the refactor a few PRs ago. 

Would you prefer if I made an issue to highlight this? Feels minor, but could lead to problems if misused in the future.